src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -77,62 +89,77 @@ void WriteBufferFromAzureBlobStorage::execWithRetry(std::function<void()> func,
 
 void WriteBufferFromAzureBlobStorage::finalizeImpl()
 {
+    auto block_blob_client = blob_container_client->GetBlockBlobClient(blob_path);
+
+    if (buffer_allocation_policy->getBufferNumber() == 1)
```

Comment:
Please add comments. It's not really obvious what this block is doing...

Reply:
added a comment